### ISSUE TITLE
The API is not consistent in the return types it provides

### ISSUE THREAD

**Issue by fguitton** [dsi-icl/optimise-core#35]
API is configured to answer to respond with JSON data. A LOT of the API calls are sending raw flat text as a response, this is not good.

**Comment by Radiateurs** [comment]
Are you talking about the error messages mostly ?

**Comment by sou-c** [comment]
also have this ques

**Comment by fguitton** [comment]
@Radiateurs @sou-c Yes. Something like the following would be more appropriate:

```
{
    error: ERROR_CODE,
    message: "Verbose error message",
    status: [404/500/...],
    stack: [Underlying_Error_Object]
}
```

This is a more powerful approach that can cope with deeply nested error inspection at dev time and greater control of error bleed and intercept. Also, as mentioned before, this allows you to maintain the API contract to return JSON at all time. Both **borderline** and **eae** exploit such mechanism; see `ErrorHelper` and `ErrorStack` classes.

**Comment by Radiateurs** [comment]
[Update] Fixed in 661718596db73a9575a8f5f9eefd47e90fa958fc
